File: java/Arrays/48_Rotate_Image.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
class RotateImage{
     public void rotate(int[][] matrix) {
        int temp;
        int n = matrix.length;
        for(int i = 0 ; i < n ; i++){
            for(int j = i+1 ; j <n ; j++){
                temp = matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        for(int i = 0 ; i < n ; i++){
            for (int j =0;j < n/2 ;j++){
                temp= matrix[i][j];
                matrix[i][j]=matrix[i][n-j-1];
                matrix[i][n-j-1]=temp;
            }
        }
        for(int i = 0; i < matrix.length; i++) {
    System.out.println(Arrays.toString(matrix[i]));
}
        
    }
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of rows : ");
        int n = sc.nextInt();
        System.out.println("Enter elements one by one : ");
        int[][] matrix = new int[n][n];
        for(int i = 0 ; i < n ; i++){
            for(int j = 0 ; j <n ; j++){
                matrix[i][j]=sc.nextInt();
            }
        }
        RotateImage obj = new RotateImage();
        obj.rotate(matrix);
    }
}

File: python/Arrays/48_Rotate_Image.py
class RotateImage(object):
    def rotate(self, matrix):
        n = len(matrix)
        for i in range(n):
            for j in range(i+1,n):
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
        for i in range(n):
            for j in range(n//2):
                matrix[i][j],matrix[i][n-j-1]=matrix[i][n-j-1],matrix[i][j]
        return matrix

        
def main():
    matrix = eval(input("Enter the matrix : "))
    obj = RotateImage()
    print(obj.rotate(matrix))

if __name__ == "__main__":
    main()